src/components/_app/IntercomProvider.tsx

```diff
@@ -89,21 +62,90 @@ const IntercomProvider = ({ children }: PropsWithChildren<unknown>): JSX.Element
         (guild1, guild2) => guild2.memberCount - guild1.memberCount
       )[0]?.memberCount,
     })
-  }, [address, isWeb3Connected, user, yourGuilds])
+  }, [address, isWeb3Connected, connectorName, walletType, user, yourGuilds])
+}
 
-  return (
-    <IntercomContext.Provider
-      value={{
-        addIntercomSettings,
-        triggerChat,
-      }}
-    >
-      {children}
-    </IntercomContext.Provider>
+type IntercomTriggerProps = {
+  "data-intercom-selector": string
+}
+
+const IntercomTrigger = (props: PropsWithChildren<IntercomTriggerProps>) => {
+  const ref = useRef<HTMLDivElement>(null)
+  const [hasClicked, setHasClicked] = useState(false)
+  const [state, loadChat] = useChat()
+
+  useEffect(() => {
+    if (!hasClicked || state !== "complete") return
+    ref.current?.click()
+    setHasClicked(false)
+  }, [hasClicked, state])
+
+  const onClick = () => {
+    if (!window.Intercom) {
+      setHasClicked(true)
+      loadChat({ open: false })
+    }
+  }
+
+  return <Box ref={ref} onClick={onClick} {...props} />
+}
+
+const triggerChat = () => {
+  if (typeof window === "undefined") return
+
+  if (window.Intercom) {
+    window.Intercom?.("show")
+    return
+  }
+
+  const facade: HTMLButtonElement = document.querySelector(
+    ".live-chat-loader-placeholder [role='button']"
+  )
+  facade?.click()
+}
+
+const addIntercomSettings = (newData: Window["intercomSettings"]) => {
+  if (typeof window === "undefined" || !newData) return
+
+  if (!window.intercomSettings) window.intercomSettings = {}
+
+  const shouldUpdate = Object.entries(newData).some(
+    ([key, value]) => window.intercomSettings[key] !== value
   )
+
+  if (!shouldUpdate) return
+
+  window.intercomSettings = {
+    ...window.intercomSettings,
+    ...newData,
+  }
+
+  // In case Intercom is loaded, we update the setting
+  window.Intercom?.("update", window.intercomSettings)
 }
 
-const useIntercom = () => useContext(IntercomContext)
+const pushToIntercomSetting = (settingName: string, value: string) => {
+  if (typeof window === "undefined") return
+
+  if (!window.intercomSettings) window.intercomSettings = {}
+
+  if (window.intercomSettings[settingName]?.toString().length)
+    window.intercomSettings[settingName] += `,${value}`
+  else window.intercomSettings[settingName] = value
+
+  // In case Intercom is loaded, we update the setting
+  window.Intercom?.("update", window.intercomSettings)
+}
+
+declare global {
+  interface Window {
+    intercomSettings?: Record<string, string | number | boolean>
+    Intercom?: (
+      action: "show" | "update",
```

Comment:
Only specified the actions which we actually use ([source](https://developers.intercom.com/installing-intercom/web/methods))